src/package_json.rs

```diff
@@ -125,85 +64,125 @@ impl PackageJson {
         json: &str,
         options: &ResolveOptions,
     ) -> Result<Self, serde_json::Error> {
-        let mut package_json_value: serde_json::Value = serde_json::from_str(json)?;
-        let mut package_json: Self = Self::deserialize(&package_json_value)?;
+        let mut data: BasePackageJson = serde_json::from_str(json)?;
+
+        let mut package_json = Self::default();
         package_json.main_fields.reserve_exact(options.main_fields.len());
         package_json.browser_fields.reserve_exact(options.alias_fields.len());
         package_json.exports.reserve_exact(options.exports_fields.len());
 
-        if let Some(package_json_value) = package_json_value.as_object_mut() {
-            // Dynamically create `main_fields`.
-            for main_field_key in &options.main_fields {
-                // Using `get` + `clone` instead of remove here
-                // because `main_fields` may contain `browser`, which is also used in `browser_fields.
-                if let Some(serde_json::Value::String(value)) =
-                    package_json_value.get(main_field_key)
-                {
-                    package_json.main_fields.push(value.clone());
+        // Dynamically create `main_fields`.
+        for main_field_key in &options.main_fields {
+            match main_field_key.as_str() {
```

Comment:
Since the crate has explicit fields for this, had to use a match: https://docs.rs/nodejs_package_json/0.1.1/nodejs_package_json/struct.PackageJson.html#structfield.main